src/services/player_services.ts

```diff
@@ -0,0 +1,65 @@
+import { CollectionReference, DocumentData } from "@google-cloud/firestore";
+import { User } from "discord.js";
+import { FirestoreDB } from "../database/firestore";
+
+export class PlayerInfo {
+    avatar: string;
+    avatarURL?: string | null;
+    id: string;
+    username: string;
+
+    constructor(id: string, data: DocumentData) {
+        this.avatar = data.avatar;
+        this.avatarURL = data.avatarURL ? data.avatarURL : undefined;
+        this.id = id;
+        this.username = data.username;
+    }
+}
+
+
+export class PlayerServices {
+    player_collection: CollectionReference;
+
+    constructor(firestore_db: FirestoreDB) {
+        this.player_collection = firestore_db.firestore.collection("players");
+    }
+
+    add_player(user: User) {
+        try {
+            this.player_collection.doc(user.id).set({
+                avatar: user.avatar,
+                avatarURL: user.avatarURL,
+                username: user.username
+            });
+        }
+        catch (err) {
+            console.error("Error adding player to Firestore", err);

```

Comment:
we should have an error util class that passes that message back to the chat room

Reply:
for all error catches